### PR TITLE
feat: implement station production view & assembly station view KAN-34 & KAN-55

### PR DESCRIPTION
**Pull request by M-Weirauch**
## Overview

Please provide a brief description of what this pull request aims to achieve.
Closes KAN-34 / closes kan-34
Closes KAN-55 / closes kan-55
Closes KAN-127 / closes kan-127
Closes KAN-131 / closes kan-131
Closes KAN-113 / closes kan-113
Closes KAN-23 / closes kan-23

### Related Jira Story

KAN-34: https://simonsays-zhaw.atlassian.net/browse/KAN-34?atlOrigin=eyJpIjoiOGZjYWE4NTIxYmYyNDY4NWE5NjYwMmYxNzhkMjU4MzQiLCJwIjoiaiJ9

KAN-55: https://simonsays-zhaw.atlassian.net/browse/KAN-55?atlOrigin=eyJpIjoiOGVjNzY4Yzc1NmZiNDE2ODk4Yjg4OTM5ZmYwZGM1M2QiLCJwIjoiaiJ9

### Backend Changes

- Added Station Production View
- Ingredients can be marked as done at a station
- Added Assembly Station View
- Orders / Menu Items / Menus can be marked as done at the assembly station
- Added Sockets for dynamic updating of the views

### Frontend Changes

- Summary of frontend changes here

## Definition of Done

- [ ] The feature must be fully tested and all specified requirements being met as outlined in the feature description.
- [ ] Unit and integrations test must be written for any business logic.
- [ ] Sufficient negative tests must be included to assess the robustness and error handling capabilities of the code.
- [ ] The pull request must be reviewed and approved by at least one codeowner of the modified files.
- [ ] All pipelines on the pull request succeed.
- [ ] Refactoring on the code has taken place.

---

Please review the above changes. Upon approval, please squash the commits following the guidelines before merging to the main branch.

### CODE REVIEW

**Review by koller-patrick — APPROVED** [review]

File: backend/src/main/kotlin/ch/zhaw/pm4/simonsays/service/StationService.kt [hunk 1]
```diff
@@ -1,12 +1,117 @@
...
+
+            if(foodOrder.menus!!.isNotEmpty() || foodOrder.menuItems!!.isNotEmpty()) {
+                processedOrders.add(foodOrder)
+            }
+
+        }
+        return processedOrders.map { orderMapper.mapOrderToOrderDTO(it) }
+    }
+
+    private fun makeStationReadyForUpdate(station: StationCreateUpdateDTO, eventId: Long, ingredients: List<Ingredient>): Station {
```
Comment (larawaespe) [inline backend/src/main/kotlin/ch/zhaw/pm4/simonsays/service/StationService.kt:90]: please move the private fun to the end of the file

**Review by larawaespe — CHANGES_REQUESTED** [review]

File: backend/src/main/resources/application-dev.yaml [hunk 2]
```diff
@@ -14,13 +14,13 @@ debug: on
 application:
   url: "http://localhost:3000"
   adminToken: "mysecretpassword"
+  frontend-origins:
```
Comment (Mcluky) [inline backend/src/main/resources/application-dev.yaml:17]: can you update this for the other yaml files as well?

**Review by Mcluky — CHANGES_REQUESTED** [review]

File: backend/src/main/resources/application-dev.yaml [hunk 3]
```diff
@@ -14,13 +14,13 @@ debug: on
 application:
   url: "http://localhost:3000"
```
Comment (Mcluky) [inline backend/src/main/resources/application-dev.yaml:15]: was this not also used just for cors? Can we not remove this when we now have the other configuration field for the frontend origins?

Comment (M-Weirauch) [reply backend/src/main/resources/application-dev.yaml]: @Mcluky I cleaned up the config files a bit, can you check to makre sure I didn't break anything?

**Review by larawaespe — APPROVED** [review]

**Review by Mcluky — APPROVED** [review]

**Pull request merged**